h/static/scripts/time.js

```diff
@@ -6,43 +6,115 @@ var day = hour * 24;
 var month = day * 30;
 var year = day * 365;
 
+function lessThanThirtySecondsAgo(date, now) {
+  return ((now - date) < 30000);
+}
+
+function lessThanOneMinuteAgo(date, now) {
+  return ((now - date) < 60000);
+}
+
+function lessThanOneHourAgo(date, now) {
+  return ((now - date) < (60 * 60 * 1000));
+}
+
+function lessThanOneDayAgo(date, now) {
+  return ((now - date) < (24 * 60 * 60 * 1000));
+}
+
+function thisYear(date, now) {
+  return date.getFullYear() === now.getFullYear();
+}
+
+function delta(date, now) {
+  return Math.round((now - date) / 1000);
+}
+
+function nSec(date, now) {
+  return '{} sec'.replace('{}', Math.floor(delta(date, now)));
+}
+
+function nMin(date, now) {
+  return '{} min'.replace('{}', Math.floor(delta(date, now) / minute));
+}
+
+function nHr(date, now) {
+  return '{} hr'.replace('{}', Math.floor(delta(date, now) / hour));
+}
+
+// Cached DateTimeFormat instances,
+// because instantiating a DateTimeFormat is expensive.
```

Comment:
Clarification here - instantiating a DateTimeFormat was very expensive in the browsers I tested and `Date.toLocaleDateTime()` which is the obvious thing to use in this context appears to do the same thing behind the scenes.

Something I have seen occasionally in the Angular code which I quite like is a link to a benchmark which can easily be run. JSPerf used to be popular for this but the site is currently being rebuilt. An alternative would be a link to a simple Gist which we can easily run in future.